example/bufferCopy/src/bufferCopy.cpp

```diff
@@ -32,72 +32,55 @@ ALPAKA_FN_ACC size_t linIdxToPitchedIdx(size_t const globalIdx, size_t const pit
 //! Prints all elements of the buffer.
 struct PrintBufferKernel
 {
-    template<typename TAcc, typename TData, typename TExtent>
+    template<typename TAcc, typename TMemoryHandle, typename TData, typename TIdx>
     ALPAKA_FN_ACC auto operator()(
         TAcc const& acc,
-        TData const* const buffer,
-        TExtent const& extents,
-        size_t const pitch) const -> void
+        alpaka::Accessor<TMemoryHandle, TData, TIdx, 3, alpaka::ReadAccess> const data) const -> void
```

Comment:
I still feel that having `TMemoryHandle` in the interface is unnecessary for the default case. We should introduce some convenience typedefs that can be specialized for different accelerators, like so:

```c++
// General case
template <typename TAcc, typename TData, typename TIdx, std::size_t TDim, typename TMode>
using BufferAccessor = Accessor<TData*, TData, TIdx, TDim, TMode>;

// Special case
template <typename TData, typename TIdx, std::size_t TDim, typename TMode>
using BufferAccessor<AccGenericSycl, TData, TIdx, TDim, TMode> = Accessor<sycl::accessor</* ... */>, TData, TIdx, TDim, TMode>;

// Usage in kernel interface
BufferAccessor<TAcc, TData, TIdx, 3, alpaka::ReadAccess> data
```

On a side note, is it still necessary to make read-only accessors `const`?

Reply:
Would this work for you?

```c++
    template<typename TAcc, typename TElem, typename TIdx, std::size_t TDim, typename TAccessModes = ReadWriteAccess>
    using BufferAccessor = Accessor<MemoryHandle<Buf<TAcc, TElem, DimInt<TDim>, TIdx>>, TElem, TIdx, TDim, TAccessModes>;
```

We then need to specialize `MemoryHandle` for each Buffer type.

Reply:
> On a side note, is it still necessary to make read-only accessors const?

Accessors are similar to `std::span` and can be used fully with or without being `const`.

Reply:
> Would this work for you?

I believe it would, yes.

Reply:
> > Would this work for you?
> 
> I believe it would, yes.

I think it cannot work if your memory handle type also depends on the access mode. I will go via `accessWith` as well.